### PR TITLE
Migrate Taskdetails API to RTK Query

### PR DESCRIPTION
**Pull request by kotesh-arya**
- Added a custom hook using RTK query which is a replacement for the API calls of TaskDetails.
 - Added Tests for the custom hook.

### CODE REVIEW

File: src/interfaces/taskDetails.type.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export type taskDetailsDataType = {
```
Comment (bhtibrewal) [inline src/interfaces/taskDetails.type.ts:1]: I think this should  be an interface

File: src/components/taskDetails/index.tsx [hunk 2]
```diff
@@ -70,15 +70,15 @@ const TaskDetails: FC<Props> = ({ url, taskID }) => {
...
+    const { data, isError, isLoading } = useGetTaskDetailsQuery(taskID);
     const { SUCCESS, ERROR } = ToastTypes;
     const { taskDetails } = state;
     useEffect(() => {
-        const fetchedData: task = { ...response.taskData };
+        const fetchedData = data?.taskData;
         dispatch({ type: 'setTaskDetails', payload: fetchedData });
         initialDataRef.current = fetchedData;
-    }, [isLoading, response]);
+    }, [isLoading, data]);
```
[the remark this answers is not in the document]
Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx:81]: In this case, the Superuser clicks `cancel` button, then I am resetting the title to initial-state by accessing the initial-data, which I am storing.

**Comment by kotesh-arya** [comment]
> I don't see the task details page test that was failing being fixed here, are you doing that in a follow-up PR?

Yes, I have created a dedicated ticket for that and will address the tests in a follow-up PR
Issue Ticket: https://github.com/Real-Dev-Squad/website-status/issues/504

File: src/interfaces/taskDetails.type.ts (hunk 1 above)
Comment (kotesh-arya) [reply src/interfaces/taskDetails.type.ts:1]: As I am **NOT EXTENDING** this by declaring it a second time anywhere, I have used the `type` here.

File: src/app/services/api.ts [hunk 3]
```diff
@@ -14,12 +14,19 @@ export const api = createApi({
...
         'Tasks',
+        'TaskDetails',
+
         'Mine_Tasks',
         'TaskTag',
+
         'User',
+
         'Tags',
+
```
Comment (Muralidhar22) [inline src/app/services/api.ts:29]: why additional space is added?

Comment (kotesh-arya) [reply src/app/services/api.ts]: Removed them 👍🏻

File: src/components/taskDetails/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx]: **Update:** Migrated the API calls to RTK-Query completely, not using any dispatch now 👍🏻

File: src/app/services/api.ts [hunk 4]
```diff
@@ -16,6 +16,7 @@ export const api = createApi({
         'Users',
         'Status',
         'Tasks',
+        'TaskDetails',
         'Mine_Tasks',
```
Comment (bhtibrewal) [inline src/app/services/api.ts:20]: Let's follow the same naming convention

**Review by bhtibrewal — COMMENTED** [review]
it looks okay to me, please address the comment

File: src/app/services/api.ts (hunk 4 above)
Comment (kotesh-arya) [reply src/app/services/api.ts]: Done 👍🏻

File: src/components/taskDetails/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kotesh-arya) [reply src/components/taskDetails/index.tsx]: > This was not what I was trying pointing out, here I meant why do we need to store the fetchedData in another state here?

When the superuser starts editing the task-title/task-description and If incase they don't want to save it. Then they will click on the `cancel` and then the **initially** fetched data should be shown as the task-title and task-description.
In order to access that initially fetched data, I am storing the fetchedData in another state. Which will be shown when they cancel updating.

[the remark this answers is not in the document]
Comment (bhtibrewal) [reply src/components/taskDetails/index.tsx]: Kotesh RTK not only fetches data but stores it. for that initial data, you could have directly used that data.

**Review by bhtibrewal — APPROVED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]
Great Work @kotesh-arya

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**